astropy/table/table.py

```diff
@@ -2021,6 +2021,70 @@ def group_by(self, keys):
 
         return groups.table_group_by(self, keys)
 
+    def to_pandas(self):
+        """
+        Return a :class:`pandas.DataFrame` instance
+
+        Returns
+        -------
+        dataframe : :class:`pandas.DataFrame`
+            A pandas :class:`pandas.DataFrame` instance
+        """
+        from pandas import DataFrame
+
+        if self.has_mixin_columns:
+            raise ValueError("Cannot convert a table with mixin columns to a pandas DataFrame")
+
+        for column in self.columns:
```

Comment:
Rather than check beforehand, maybe let `DataFrame` raise an error if something bad is passed in? It keeps the code cleaner, and if pandas starts to support multi-dimensional columns, things will just work. (or If you think it is important to give a clearer error message, maybe have the actual data frame assignments inside try/except clauses?).